src/main/java/com/gluonhq/substrate/util/FileDeps.java

```diff
@@ -231,10 +234,9 @@ private boolean setupDependencies() throws IOException {
                 downloadJavaFXStatic = true;
             } else {
                 String path = javafxStatic.toString();
-                if (JAVAFX_FILES.stream()
-                        .map(s -> new File(path, s))
-                        .anyMatch(f -> !f.exists())) {
-                    Logger.logDebug("jar file not found");
+                if (JAVAFX_FILES.stream().map(s -> new File(path, s)).anyMatch(f -> !f.exists()) &&
```

Comment:
if the check for the JavaFX jars is true, means that some jar is missing, therefore: `downloadJavaFXStatic = true;`
Adding "and either libglassXX.a"  doesn't help. Shouldn't it be `noneMatch`?

```
if (JAVAFX_FILES.stream().map(s -> new File(path, s)).anyMatch(f -> !f.exists()) || 
     JAVAFX_STATIC_FILES.stream().map(s -> new File(path, s)).noneMatch(File::exists)) {
```

Reply:
Have you checked #489 for explanation?

Reply:
Yes, I did. 
We used the javafx list to verify that those files are available or not. If any of them is missing, we download.
Now we need to split that in two: jars and libs. Okay.

But your code is doing:
 - if any jar is missing and either lib exists -> download

so at this point, if a jar is missing but one lib exist, it won't download. Or if all jars exist, but no lib is found, it won't download.

Imo, the check should be:
- if any jar is missing -> download, or
- if none of these libs is found -> download

Reply:
Makes sense. Thank you for the explanation.